src/opendr/perception/pose_estimation/hr_pose_estimation/HighResolutionLearner.py

```diff
@@ -0,0 +1,992 @@
+            if xmax + extra_pad_x < img.shape[1]:
+                xmax = xmax + extra_pad_x
+            if ymin - extra_pad_y > 0:
+                ymin = ymin - extra_pad_y
+            if ymax + extra_pad_y < img.shape[0]:
+                ymax = ymax + extra_pad_y
+
+            if (xmax - xmin) > 40 and (ymax - ymin) > 40:
+                crop_img = img[ymin:ymax, xmin:xmax]
+            else:
+                crop_img = img[offset:img.shape[0], offset:img.shape[1]]
+
+            h, w, _ = crop_img.shape
+
+            # ------- Second pass of the image, inference for pose estimation -------
+            avg_heatmaps, avg_pafs, scale, pad = \
+                HighResolutionPoseEstimationLearner.second_pass_infer(self, self.model, crop_img,
+                                                                      self.second_pass_height,
+                                                                      max_width, stride, upsample_ratio)
+
+            total_keypoints_num = 0
+            all_keypoints_by_type = []
+            for kpt_idx in range(18):
+                total_keypoints_num += extract_keypoints(avg_heatmaps[:, :, kpt_idx], all_keypoints_by_type,
+                                                         total_keypoints_num)
+
+            pose_entries, all_keypoints = group_keypoints(all_keypoints_by_type, avg_pafs)
+
+            for kpt_id in range(all_keypoints.shape[0]):
+                all_keypoints[kpt_id, 0] = (all_keypoints[kpt_id, 0] * stride / upsample_ratio - pad[1]) / scale
+                all_keypoints[kpt_id, 1] = (all_keypoints[kpt_id, 1] * stride / upsample_ratio - pad[0]) / scale
+
+            for i in range(all_keypoints.shape[0]):
+                for j in range(all_keypoints.shape[1]):
+                    if j == 0:  # Adjust offset if needed for evaluation on our HR datasets
+                        all_keypoints[i][j] = round((all_keypoints[i][j] + xmin) - offset)
+                    if j == 1:  # Adjust offset if needed for evaluation on our HR datasets
+                        all_keypoints[i][j] = round((all_keypoints[i][j] + ymin) - offset)
+
+            current_poses = []
+            for n in range(len(pose_entries)):
+                if len(pose_entries[n]) == 0:
+                    continue
+                pose_keypoints = np.ones((num_keypoints, 2), dtype=np.int32) * -1
+                for kpt_id in range(num_keypoints):
+                    if pose_entries[n][kpt_id] != -1.0:  # keypoint was found
+                        pose_keypoints[kpt_id, 0] = int(all_keypoints[int(pose_entries[n][kpt_id]), 0])
+                        pose_keypoints[kpt_id, 1] = int(all_keypoints[int(pose_entries[n][kpt_id]), 1])
+                pose = Pose(pose_keypoints, pose_entries[n][18])
+                current_poses.append(pose)
+
+        return current_poses
+
+    def infer_light_odr(self, img, upsample_ratio=4, track=True, smooth=True):  # LwOP from OpenDR implementation
```

Comment:
Method should probably be removed as discussed in the corresponding comments on the doc